### ISSUE TITLE
Compiler panic on missing .await when calling async fn returning T1<impl T2> 

### ISSUE THREAD

**Issue by EdMcBane** [rust-lang/rust#103889]
### Code

```Rust
trait Responder {}
impl Responder for () {}

async fn status() -> Option<impl Responder> {
    do_status()
}

async fn do_status() -> Option<impl Responder> {
    Some(())
}
```

### Meta
**Fails (rustc panics) on stable rustc 1.64.0**

`rustc --version --verbose`:
```
rustc 1.64.0 (a55dd71d5 2022-09-19)
binary: rustc
commit-hash: a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52
commit-date: 2022-09-19
host: aarch64-apple-darwin
release: 1.64.0
LLVM version: 14.0.6
```

**Works correctly (compilation fails without rustc panic) on rustc 1.67.0-nightly (11ebe6512 2022-11-01)**
```
rustc 1.67.0-nightly (11ebe6512 2022-11-01)
binary: rustc
commit-hash: 11ebe6512b4c77633c59f8dcdd421df3b79d1a9f
commit-date: 2022-11-01
host: aarch64-apple-darwin
release: 1.67.0-nightly
LLVM version: 15.0.4
```

### Error output

```
   Compiling overseer v0.1.0 (/Users/fdegrassi/projects/rust_compiler_bug_overseer)
thread 'rustc' panicked at 'index out of bounds: the len is 6 but the index is 7', /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/ena-0.14.0/src/snapshot_vec.rs:199:10
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic_bounds_check
   3: <rustc_infer::infer::type_variable::TypeVariableTable>::probe
   4: <rustc_infer::infer::ShallowResolver as rustc_middle::ty::fold::TypeFolder>::fold_ty
   5: <&rustc_middle::ty::list::List<rustc_middle::ty::subst::GenericArg> as rustc_middle::ty::fold::TypeFoldable>::try_fold_with::<rustc_infer::infer::resolve::OpportunisticVarResolver>
   6: <rustc_middle::ty::Ty as rustc_middle::ty::fold::TypeSuperFoldable>::super_fold_with::<rustc_infer::infer::resolve::OpportunisticVarResolver>
   7: <rustc_infer::infer::InferCtxt>::same_type_modulo_infer
   8: <rustc_infer::infer::InferCtxt>::note_type_err
   9: <rustc_infer::infer::InferCtxt>::report_and_explain_type_error
  10: <rustc_infer::infer::InferCtxt>::report_mismatched_types
  11: <rustc_typeck::check::coercion::CoerceMany<&rustc_hir::hir::Expr>>::report_return_mismatched_types
  12: <rustc_typeck::check::coercion::CoerceMany<&rustc_hir::hir::Expr>>::coerce_inner
  13: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_block_with_expected
  14: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_expr_kind
  15: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  16: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_expr_kind
  17: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  18: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_block_with_expected
  19: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_expr_kind
  20: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  21: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_return_expr
  22: rustc_typeck::check::check::check_fn
  23: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_expr_closure
  24: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_expr_kind
  25: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  26: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_argument_types
  27: <rustc_typeck::check::fn_ctxt::FnCtxt>::confirm_builtin_call
  28: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_call
  29: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_expr_kind
  30: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  31: <rustc_typeck::check::fn_ctxt::FnCtxt>::check_return_expr
  32: rustc_typeck::check::check::check_fn
  33: <rustc_infer::infer::InferCtxtBuilder>::enter::<&rustc_middle::ty::context::TypeckResults, <rustc_typeck::check::inherited::InheritedBuilder>::enter<rustc_typeck::check::typeck_with_fallback<rustc_typeck::check::typeck::{closure#0}>::{closure#1}, &rustc_middle::ty::context::TypeckResults>::{closure#0}>
  34: rustc_typeck::check::typeck
  35: <rustc_query_system::dep_graph::graph::DepGraph<rustc_middle::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle::ty::context::TyCtxt, rustc_span::def_id::LocalDefId, &rustc_middle::ty::context::TypeckResults>
  36: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::plumbing::QueryCtxt, rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::LocalDefId, &rustc_middle::ty::context::TypeckResults>>
  37: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::typeck, rustc_query_impl::plumbing::QueryCtxt>
  38: <rustc_middle::ty::context::TyCtxt>::typeck_opt_const_arg
  39: rustc_mir_build::build::mir_built
  40: <rustc_middle::dep_graph::dep_node::DepKind as rustc_query_system::dep_graph::DepKind>::with_deps::<<rustc_query_system::dep_graph::graph::DepGraph<rustc_middle::dep_graph::dep_node::DepKind>>::with_task_impl<rustc_middle::ty::context::TyCtxt, rustc_middle::ty::WithOptConstParam<rustc_span::def_id::LocalDefId>, &rustc_data_structures::steal::Steal<rustc_middle::mir::Body>>::{closure#0}, &rustc_data_structures::steal::Steal<rustc_middle::mir::Body>>
  41: <rustc_query_system::dep_graph::graph::DepGraph<rustc_middle::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle::ty::context::TyCtxt, rustc_middle::ty::WithOptConstParam<rustc_span::def_id::LocalDefId>, &rustc_data_structures::steal::Steal<rustc_middle::mir::Body>>
  42: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::plumbing::QueryCtxt, rustc_query_system::query::caches::DefaultCache<rustc_middle::ty::WithOptConstParam<rustc_span::def_id::LocalDefId>, &rustc_data_structures::steal::Steal<rustc_middle::mir::Body>>>
  43: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::mir_built, rustc_query_impl::plumbing::QueryCtxt>
  44: <rustc_query_impl::Queries as rustc_middle::ty::query::QueryEngine>::mir_built
  45: rustc_mir_transform::check_unsafety::unsafety_check_result
  46: <rustc_mir_transform::check_unsafety::provide::{closure#0} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, rustc_span::def_id::LocalDefId)>>::call_once
  47: <rustc_query_system::dep_graph::graph::DepGraph<rustc_middle::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle::ty::context::TyCtxt, rustc_span::def_id::LocalDefId, &rustc_middle::mir::query::UnsafetyCheckResult>
  48: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::plumbing::QueryCtxt, rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::LocalDefId, &rustc_middle::mir::query::UnsafetyCheckResult>>
  49: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::unsafety_check_result, rustc_query_impl::plumbing::QueryCtxt>
  50: rustc_mir_transform::mir_const
  51: <rustc_middle::dep_graph::dep_node::DepKind as rustc_query_system::dep_graph::DepKind>::with_deps::<<rustc_query_system::dep_graph::graph::DepGraph<rustc_middle::dep_graph::dep_node::DepKind>>::with_task_impl<rustc_middle::ty::context::TyCtxt, rustc_middle::ty::WithOptConstParam<rustc_span::def_id::LocalDefId>, &rustc_data_structures::steal::Steal<rustc_middle::mir::Body>>::{closure#0}, &rustc_data_structures::steal::Steal<rustc_middle::mir::Body>>
  52: <rustc_query_system::dep_graph::graph::DepGraph<rustc_middle::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle::ty::context::TyCtxt, rustc_middle::ty::WithOptConstParam<rustc_span::def_id::LocalDefId>, &rustc_data_structures::steal::Steal<rustc_middle::mir::Body>>
  53: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::plumbing::QueryCtxt, rustc_query_system::query::caches::DefaultCache<rustc_middle::ty::WithOptConstParam<rustc_span::def_id::LocalDefId>, &rustc_data_structures::steal::Steal<rustc_middle::mir::Body>>>
  54: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::mir_const, rustc_query_impl::plumbing::QueryCtxt>
  55: <rustc_query_impl::Queries as rustc_middle::ty::query::QueryEngine>::mir_const
  56: rustc_mir_transform::mir_promoted
  57: <rustc_middle::dep_graph::dep_node::DepKind as rustc_query_system::dep_graph::DepKind>::with_deps::<<rustc_query_system::dep_graph::graph::DepGraph<rustc_middle::dep_graph::dep_node::DepKind>>::with_task_impl<rustc_middle::ty::context::TyCtxt, rustc_middle::ty::WithOptConstParam<rustc_span::def_id::LocalDefId>, (&rustc_data_structures::steal::Steal<rustc_middle::mir::Body>, &rustc_data_structures::steal::Steal<rustc_index::vec::IndexVec<rustc_middle::mir::Promoted, rustc_middle::mir::Body>>)>::{closure#0}, (&rustc_data_structures::steal::Steal<rustc_middle::mir::Body>, &rustc_data_structures::steal::Steal<rustc_index::vec::IndexVec<rustc_middle::mir::Promoted, rustc_middle::mir::Body>>)>
  58: <rustc_query_system::dep_graph::graph::DepGraph<rustc_middle::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle::ty::context::TyCtxt, rustc_middle::ty::WithOptConstParam<rustc_span::def_id::LocalDefId>, (&rustc_data_structures::steal::Steal<rustc_middle::mir::Body>, &rustc_data_structures::steal::Steal<rustc_index::vec::IndexVec<rustc_middle::mir::Promoted, rustc_middle::mir::Body>>)>
  59: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::mir_promoted, rustc_query_impl::plumbing::QueryCtxt>
  60: <rustc_query_impl::Queries as rustc_middle::ty::query::QueryEngine>::mir_promoted
  61: rustc_borrowck::mir_borrowck
  62: <rustc_borrowck::provide::{closure#0} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, rustc_span::def_id::LocalDefId)>>::call_once
  63: <rustc_query_system::dep_graph::graph::DepGraph<rustc_middle::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle::ty::context::TyCtxt, rustc_span::def_id::LocalDefId, &rustc_middle::mir::query::BorrowCheckResult>
  64: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::plumbing::QueryCtxt, rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::LocalDefId, &rustc_middle::mir::query::BorrowCheckResult>>
  65: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::mir_borrowck, rustc_query_impl::plumbing::QueryCtxt>
  66: rustc_typeck::collect::type_of::find_opaque_ty_constraints_for_rpit
  67: rustc_typeck::collect::type_of::type_of
  68: <rustc_query_system::dep_graph::graph::DepGraph<rustc_middle::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle::ty::context::TyCtxt, rustc_span::def_id::DefId, rustc_middle::ty::Ty>
  69: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::plumbing::QueryCtxt, rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::DefId, rustc_middle::ty::Ty>>
  70: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::type_of, rustc_query_impl::plumbing::QueryCtxt>
  71: rustc_typeck::check::check::check_opaque
  72: rustc_typeck::check::check::check_item_type
  73: rustc_typeck::check::check::check_mod_item_types
  74: <rustc_query_system::dep_graph::graph::DepGraph<rustc_middle::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle::ty::context::TyCtxt, rustc_span::def_id::LocalDefId, ()>
  75: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::plumbing::QueryCtxt, rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::LocalDefId, ()>>
  76: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::check_mod_item_types, rustc_query_impl::plumbing::QueryCtxt>
  77: <rustc_session::session::Session>::time::<(), rustc_typeck::check_crate::{closure#6}>
  78: rustc_typeck::check_crate
  79: rustc_interface::passes::analysis
  80: <rustc_query_system::dep_graph::graph::DepGraph<rustc_middle::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle::ty::context::TyCtxt, (), core::result::Result<(), rustc_errors::ErrorGuaranteed>>
  81: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::plumbing::QueryCtxt, rustc_query_system::query::caches::DefaultCache<(), core::result::Result<(), rustc_errors::ErrorGuaranteed>>>
  82: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::analysis, rustc_query_impl::plumbing::QueryCtxt>
  83: <rustc_interface::passes::QueryContext>::enter::<rustc_driver::run_compiler::{closure#1}::{closure#2}::{closure#3}, core::result::Result<(), rustc_errors::ErrorGuaranteed>>
  84: <rustc_interface::interface::Compiler>::enter::<rustc_driver::run_compiler::{closure#1}::{closure#2}, core::result::Result<core::option::Option<rustc_interface::queries::Linker>, rustc_errors::ErrorGuaranteed>>
  85: rustc_span::with_source_map::<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_interface::interface::create_compiler_and_run<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>::{closure#1}>
  86: <scoped_tls::ScopedKey<rustc_span::SessionGlobals>>::set::<rustc_interface::interface::run_compiler<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>::{closure#0}, core::result::Result<(), rustc_errors::ErrorGuaranteed>>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.64.0 (a55dd71d5 2022-09-19) running on aarch64-apple-darwin

note: compiler flags: --crate-type lib -C embed-bitcode=no -C split-debuginfo=unpacked -C debuginfo=2 -C incremental

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
#0 [typeck] type-checking `status`
#1 [mir_built] building MIR for `status`
#2 [unsafety_check_result] unsafety-checking `status`
#3 [mir_const] processing MIR for `status`
#4 [mir_promoted] processing `status`
#5 [mir_borrowck] borrow-checking `status`
#6 [type_of] computing type of `status::{opaque#0}`
#7 [check_mod_item_types] checking item types in top-level module
#8 [analysis] running analysis passes on this crate
end of query stack
error: internal compiler error: no errors encountered even though `delay_span_bug` issued

error: internal compiler error: VecMap([(OpaqueTypeKey { def_id: DefId(0:6 ~ overseer[8af7]::status::{opaque#0}), substs: [] }, OpaqueTypeDecl { hidden_type: OpaqueHiddenType { span: src/lib.rs:4:22: 4:44 (#6), ty: _ }, origin: AsyncFn(DefId(0:5 ~ overseer[8af7]::status)) }), (OpaqueTypeKey { def_id: DefId(0:7 ~ overseer[8af7]::status::{opaque#0}::{opaque#0}), substs: [] }, OpaqueTypeDecl { hidden_type: OpaqueHiddenType { span: src/lib.rs:4:45: 6:2 (#0), ty: _ }, origin: FnReturn(DefId(0:5 ~ overseer[8af7]::status)) })])
  |
  = note: delayed at compiler/rustc_infer/src/infer/opaque_types/table.rs:50:26

thread 'rustc' panicked at 'Box<dyn Any>', compiler/rustc_errors/src/lib.rs:1426:13
stack backtrace:
   0:        0x1055a7978 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h1543c132bc4e188c
   1:        0x1055f4828 - core::fmt::write::hda8e8eb84b49cbfc
   2:        0x10559ac74 - std::io::Write::write_fmt::hb84c8996aec7120c
   3:        0x1055aa5d0 - std::panicking::default_hook::{{closure}}::hdf06011cb093de6a
   4:        0x1055aa334 - std::panicking::default_hook::hd7ceb942fff7b170
   5:        0x10cbf4d9c - rustc_driver[8f8caae8abf9e6c7]::DEFAULT_HOOK::{closure#0}::{closure#0}
   6:        0x1055aaba4 - std::panicking::rust_panic_with_hook::h053d4067a63a6fcb
   7:        0x110953e64 - std[dfd41530add96b74]::panicking::begin_panic::<rustc_errors[b30f6c19c09b5d30]::ExplicitBug>::{closure#0}
   8:        0x110953c40 - std[dfd41530add96b74]::sys_common::backtrace::__rust_end_short_backtrace::<std[dfd41530add96b74]::panicking::begin_panic<rustc_errors[b30f6c19c09b5d30]::ExplicitBug>::{closure#0}, !>
   9:        0x110c6564c - std[dfd41530add96b74]::panicking::begin_panic::<rustc_errors[b30f6c19c09b5d30]::ExplicitBug>
  10:        0x1109842c8 - std[dfd41530add96b74]::panic::panic_any::<rustc_errors[b30f6c19c09b5d30]::ExplicitBug>
  11:        0x11098707c - <rustc_errors[b30f6c19c09b5d30]::HandlerInner as core[470b050f2b764c47]::ops::drop::Drop>::drop
  12:        0x10cb852dc - core[470b050f2b764c47]::ptr::drop_in_place::<rustc_session[6283a40eea3c7aa9]::parse::ParseSess>
  13:        0x10cb8bf68 - <alloc[e1afe643157cdf10]::rc::Rc<rustc_session[6283a40eea3c7aa9]::session::Session> as core[470b050f2b764c47]::ops::drop::Drop>::drop
  14:        0x10cbe18a8 - core[470b050f2b764c47]::ptr::drop_in_place::<rustc_interface[efe2b5ffa23206d6]::interface::Compiler>
  15:        0x10cbdf67c - rustc_span[a44dd0315ec95fe3]::with_source_map::<core[470b050f2b764c47]::result::Result<(), rustc_errors[b30f6c19c09b5d30]::ErrorGuaranteed>, rustc_interface[efe2b5ffa23206d6]::interface::create_compiler_and_run<core[470b050f2b764c47]::result::Result<(), rustc_errors[b30f6c19c09b5d30]::ErrorGuaranteed>, rustc_driver[8f8caae8abf9e6c7]::run_compiler::{closure#1}>::{closure#1}>
  16:        0x10cb92fd0 - <scoped_tls[438d350ce76b615a]::ScopedKey<rustc_span[a44dd0315ec95fe3]::SessionGlobals>>::set::<rustc_interface[efe2b5ffa23206d6]::interface::run_compiler<core[470b050f2b764c47]::result::Result<(), rustc_errors[b30f6c19c09b5d30]::ErrorGuaranteed>, rustc_driver[8f8caae8abf9e6c7]::run_compiler::{closure#1}>::{closure#0}, core[470b050f2b764c47]::result::Result<(), rustc_errors[b30f6c19c09b5d30]::ErrorGuaranteed>>
  17:        0x10cbb243c - std[dfd41530add96b74]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[efe2b5ffa23206d6]::util::run_in_thread_pool_with_globals<rustc_interface[efe2b5ffa23206d6]::interface::run_compiler<core[470b050f2b764c47]::result::Result<(), rustc_errors[b30f6c19c09b5d30]::ErrorGuaranteed>, rustc_driver[8f8caae8abf9e6c7]::run_compiler::{closure#1}>::{closure#0}, core[470b050f2b764c47]::result::Result<(), rustc_errors[b30f6c19c09b5d30]::ErrorGuaranteed>>::{closure#0}, core[470b050f2b764c47]::result::Result<(), rustc_errors[b30f6c19c09b5d30]::ErrorGuaranteed>>
  18:        0x10cbb2af8 - <<std[dfd41530add96b74]::thread::Builder>::spawn_unchecked_<rustc_interface[efe2b5ffa23206d6]::util::run_in_thread_pool_with_globals<rustc_interface[efe2b5ffa23206d6]::interface::run_compiler<core[470b050f2b764c47]::result::Result<(), rustc_errors[b30f6c19c09b5d30]::ErrorGuaranteed>, rustc_driver[8f8caae8abf9e6c7]::run_compiler::{closure#1}>::{closure#0}, core[470b050f2b764c47]::result::Result<(), rustc_errors[b30f6c19c09b5d30]::ErrorGuaranteed>>::{closure#0}, core[470b050f2b764c47]::result::Result<(), rustc_errors[b30f6c19c09b5d30]::ErrorGuaranteed>>::{closure#1} as core[470b050f2b764c47]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  19:        0x1055b33dc - std::sys::unix::thread::Thread::new::thread_start::h403ab16d5f453cd4
  20:        0x1914ec26c - __pthread_deallocate

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.64.0 (a55dd71d5 2022-09-19) running on aarch64-apple-darwin

note: compiler flags: --crate-type lib -C embed-bitcode=no -C split-debuginfo=unpacked -C debuginfo=2 -C incremental

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
thread panicked while panicking. aborting.
error: could not compile `overseer`

Caused by:
  process didn't exit successfully: `rustc --crate-name overseer --edition=2021 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C split-debuginfo=unpacked -C debuginfo=2 -C metadata=8172757fbd5c4a86 -C extra-filename=-8172757fbd5c4a86 --out-dir /Users/fdegrassi/projects/rust_compiler_bug_overseer/target/debug/deps -C incremental=/Users/fdegrassi/projects/rust_compiler_bug_overseer/target/debug/incremental -L dependency=/Users/fdegrassi/projects/rust_compiler_bug_overseer/target/debug/deps` (signal: 6, SIGABRT: process abort signal)
```

**Comment by compiler-errors** [comment]
This will be fixed in Rust 1.65

**Comment by jackh726** [comment]
Closing since this works on nightly
